File: backend/app/schemas/billing.py
from pydantic import BaseModel, Field


class BillingPlanDto(BaseModel):
    code: str
    name: str
    priceLabel: str
    priceCents: int = 0
    chatLimit: int | None = None
    knowledgeBaseLimit: int | None = None
    features: list[str] = Field(default_factory=list)
    highlighted: bool = False


class BillingUsageDto(BaseModel):
    yearMonth: str
    used: int
    limit: int | None = None
    remaining: int | None = None


class BillingMeDto(BaseModel):
    planCode: str
    planName: str
    planUpgradedAt: str | None = None
    usage: BillingUsageDto
    canManageKnowledge: bool = True
    canUseHarnessAgent: bool = False
    knowledgeBaseLimit: int | None = None
    knowledgeBaseUsed: int = 0
    canCreateKnowledgeBase: bool = True


class CreateBillingOrderBody(BaseModel):
    planCode: str = "pro"


class BillingOrderDto(BaseModel):
    id: str
    planCode: str
    amountCents: int
    status: str
    channel: str = "alipay"
    providerTradeNo: str | None = None
    createdAt: str
    paidAt: str | None = None


class BillingPayOptionsDto(BaseModel):
    alipaySandbox: bool = True
    alipayReady: bool = False
    mockPayEnabled: bool = True
    gateway: str = ""
    stripeReady: bool = False
    stripeMode: str = "unknown"
    stripeCurrency: str = "cny"


class AlipayPagePayDto(BaseModel):
    orderId: str
    payUrl: str
    gateway: str


class StripeCheckoutDto(BaseModel):
    orderId: str
    payUrl: str
    sessionId: str
    mode: str


class UsageTrendPointDto(BaseModel):
    date: str
    chatCount: int
    activeUsers: int = 0


class UsageUserRankDto(BaseModel):
    userId: str
    username: str
    fullName: str
    chatCount: int


class AdminUsageStatsDto(BaseModel):
    totalChats: int
    activeUsers: int
    freeUsers: int
    proUsers: int
    trend: list[UsageTrendPointDto] = Field(default_factory=list)
    topUsers: list[UsageUserRankDto] = Field(default_factory=list)

File: webadmin/src/lib/api/billing.ts
import { apiFetch, readEnvelope } from './client'

export type BillingPlan = {
  code: string
  name: string
  priceLabel: string
  priceCents: number
  chatLimit: number | null
  knowledgeBaseLimit?: number | null
  features: string[]
  highlighted: boolean
}

export type BillingUsage = {
  yearMonth: string
  used: number
  limit: number | null
  remaining: number | null
}

export type BillingMe = {
  planCode: string
  planName: string
  planUpgradedAt: string | null
  usage: BillingUsage
  canManageKnowledge: boolean
  canUseHarnessAgent: boolean
  knowledgeBaseLimit?: number | null
  knowledgeBaseUsed?: number
  canCreateKnowledgeBase?: boolean
}

export type BillingOrder = {
  id: string
  planCode: string
  amountCents: number
  status: string
  channel?: string
  providerTradeNo?: string | null
  createdAt: string
  paidAt: string | null
}

export type BillingPayOptions = {
  alipaySandbox: boolean
  alipayReady: boolean
  mockPayEnabled: boolean
  gateway: string
  stripeReady: boolean
  stripeMode: string
  stripeCurrency: string
}

export type AlipayPagePay = {
  orderId: string
  payUrl: string
  gateway: string
}

export type StripeCheckout = {
  orderId: string
  payUrl: string
  sessionId: string
  mode: string
}

export type UsageTrendPoint = {
  date: string
  chatCount: number
  activeUsers: number
}

export type UsageUserRank = {
  userId: string
  username: string
  fullName: string
  chatCount: number
}

export type AdminUsageStats = {
  totalChats: number
  activeUsers: number
  freeUsers: number
  proUsers: number
  trend: UsageTrendPoint[]
  topUsers?: UsageUserRank[]
}

export async function listBillingPlans(): Promise<BillingPlan[]> {
  const res = await apiFetch('GET', '/api/v1/billing/plans')
  return readEnvelope<BillingPlan[]>(res)
}

export async function getBillingMe(): Promise<BillingMe> {
  const res = await apiFetch('GET', '/api/v1/billing/me')
  return readEnvelope<BillingMe>(res)
}

export async function createBillingOrder(planCode = 'pro'): Promise<BillingOrder> {
  const res = await apiFetch('POST', '/api/v1/billing/orders', { planCode })
  return readEnvelope<BillingOrder>(res)
}

export async function getBillingPayOptions(): Promise<BillingPayOptions> {
  const res = await apiFetch('GET', '/api/v1/billing/pay-options')
  return readEnvelope<BillingPayOptions>(res)
}

export async function startAlipayPagePay(orderId: string): Promise<AlipayPagePay> {
  const res = await apiFetch('POST', `/api/v1/billing/orders/${orderId}/alipay/page-pay`)
  return readEnvelope<AlipayPagePay>(res)
}

export async function startStripeCheckout(orderId: string): Promise<StripeCheckout> {
  const res = await apiFetch('POST', `/api/v1/billing/orders/${orderId}/stripe/checkout`)
  return readEnvelope<StripeCheckout>(res)
}

export async function mockPayBillingOrder(orderId: string): Promise<BillingOrder> {
  const res = await apiFetch('POST', `/api/v1/billing/orders/${orderId}/mock-pay`)
  return readEnvelope<BillingOrder>(res)
}

export async function getAdminUsageStats(days = 30): Promise<AdminUsageStats> {
  const res = await apiFetch('GET', `/api/v1/admin/usage-stats?days=${days}`)
  return readEnvelope<AdminUsageStats>(res)
}
